src/loader.ts

```diff
@@ -241,6 +136,53 @@ interface ModuleDefinitionArguments extends Array<any> {
 		return has;
 	})();
 
+	let listenerQueues = {};
+
+	const reportModuleLoadError = function (parent: Module, module: Module, url: string): void {
+		let parentMid = (parent ? ' (parent: ' + parent.mid + ')' : '');
+		let message = `Failed to load module ${module.mid} from ${url}${parentMid}`;
+		let error = mix<LoaderError>(new Error(message), {
+			src: 'dojo/loader',
+			info: {
+				module,
+				url,
+				parentMid
+			}
+		});
+
+		if (!emit('error', error)) { throw error; };
+	};
+
+	const emit = function(type: SignalType, args: {}): number | boolean {
+		let queue: any[] = listenerQueues[type];
+		let hasListeners = queue && queue.length;
+
+		if (hasListeners) {
+			for (let listener of queue.slice(0)) {
```

Comment:
Now I get it... ;-)  I would just add those comments there, so I don't ask silly questions again!